### ISSUE TITLE
Add support for defining subscription_id via environment variable

### ISSUE THREAD

**Issue by jerryaldrichiii** [test-kitchen/kitchen-azurerm#110]
Seems I fell into the same trap that these people did:
  - #6 
  - #72 
  - #47 

I think we should support adding a `AZURE_SUBSCRIPTION_ID` environment variable so that it need not be defined in the `.kitchen.yml` file.

I also like the idea of assuming that if there is only one entry in `~/.azure/credentials` that it is the subscription intended.

**Comment by dee3sync** [comment]
@jerryaldrichiii I do this today (running Kitchen Version 1.23.2) ...just set the `AZURE_SUBSCRIPTION_ID` as an environment variable and then add this in your test kitchen file

```
driver:
  name: azurerm
  subscription_id: "<%= ENV['AZURE_SUBSCRIPTION_ID'] %>"
```

**Comment by jerryaldrichiii** [comment]
Right, that is the only way to do it now. Since we currently set all the others via environment variables and we do not need to interpolate them in the file, I'd like to do it for `AZURE_SUBSCRIPTION_ID` too.

**Comment by stuartpreston** [comment]
Fixed in commit df79c78
